### DISCUSSION TITLE
Smart Plug v2 vs Smart plug v2 polled power

### DISCUSSION THREAD

**Discussion by Krispkiwi** [make-all/tuya-local#1009] [Q&A] [answered]
Hey mate, 

Was just trying to find information on this, what's the difference between the two? The Kogan switches appear to work with both

**Comment by make-all** [comment, marked as answer]
Some smartplugs do not return current/power/voltage readings unless they are explicitly polled for them.  These require the "polled_power" variant of the config. Some other smartplugs do not expect the explicit polling and act strangely if you use that.  If your plugs work equally fine with either config, then the smartplugv2 config will put slightly less load on your HA installation.

**Comment by Krispkiwi** [comment]
Cheers :)

**Comment by Krispkiwi** [comment]
Hey mate, just getting the new instance up now, I'll get the solar logs shortly. I notice when doing the switches there's now a "Smart Plug V2 Energy option which seems to give a few more options. Is there any documentation on that? Looks like there's a few extra options on them although not sure if they apply to my device. I'm unaware of child locks and override protections, also do you know how to tell what a devices protocol is? I've been looking in the logs and I can't see it.

**Reply by make-all** [reply]
All of the extra features in the _energy device are marked optional, as I don't want to make 100 different configs just for single smartplugs to cover all the different variations of minor extra settings they have.  In any case, if your plug needs polling, it needs to use the _polling one, but if it doesn't then if you can't see any missing features from the basic smartplug then you are only going to get extra entities that always show up as "Unavailable" by switching to the _energy config.

The device diagnostics should tell which protocol was chosen for your device if it is working in Auto mode.
